### PR TITLE
Add pyvsita 2D annulus example

### PR DESCRIPTION
**Pull request by KerrMadeleine**
Added a visualization script in pyvista that plots spherical annulus models and computes radial and angular velocities and plots them. A sample .prm to use the script on can be generated from the onset-of-convection cookbook and changed in the ways described in the bottom comments of the script. 
### Before your first pull request:

* [ X] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.

### For all pull requests:

* [X ] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).

### For new features/models or changes of existing features:

* [X ] I have tested my new feature locally to ensure it is correct.
* [ X] I have [created a testcase](https://aspect-documentation.readthedocs.io/en/latest/user/extending/testing/writing-tests.html) for the new feature/benchmark in the [tests/](../blob/main/tests/) directory.
* [X ] I have added a changelog entry in the [doc/modules/changes](../blob/main/doc/modules/changes) directory that will inform other users of my change.

### CODE REVIEW

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 1]
```diff
@@ -0,0 +1,260 @@
+# Overview (description from John Nabiloff pyvista_2D_example scripts)
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:1]: Please provide an overview of what your script does first, then at some point near the end say refer to /contrib/python/scripts/pyvista_2D_example.py) for more information about PyVista and related installation instructions.

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+# "conda install -c conda-forge pyvista". You can check
+# to see if pyvista is installed with "import pyvista as pv".
+
+# Load modules
+import numpy as np
+import pyvista as pv
+import matplotlib.pyplot as plt
+import os
+
+#   Include the path to the project directory ofyou would like to
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:44]:
```suggestion
#   Include the path to the project directory you would like to
```

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+#   visualize and list the model directories within that Input directory + model directory
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:45]:
```suggestion
#   visualize and list the model directories within that input directory + model directory
```

File: contrib/python/scripts/pyvista_2D_annulus_example.py (hunk 1 above)
Comment (naliboff) [reply contrib/python/scripts/pyvista_2D_annulus_example.py:1]: In your description of this model, please not that this script uses the time reader!

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+        "r"
+    ]
+
+    #   Define mid-mantle region, 400 km above the core-mantle boundary and
+    #   400 km below the surface to avoid the temperature effects of the the non-
+    #   mixed cold and hot thermal boundary layers on the boundaries.
+    mesh_midmantle = mesh.threshold(
+        scalars="r", value=r_inner + 400000, invert=False
+    ).threshold(scalars="r", value=r_outer - 400000, invert=True)
+    #   Compute the mean temperature value of the mid-mantle. This can be used
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:99]: Add a space above here

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 5]
```diff
@@ -0,0 +1,260 @@
...
+    #   along with the core and surface temperatures to define the thermal boundary
+    #   layers
+    mean_mantle_T = np.mean(mesh_midmantle["T"])
+    #   Create a new point field in the mesh object which is the temperature
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:103]: Add a space here for readability

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+    #   deviation from the mean at all points in the domain.
+    mesh.point_data["T_dev"] = mesh["T"] - mean_mantle_T
+
+    #    Other calculations can be done if other fields are outputed.
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:107]: I think this comment can be deleted.

**Review by naliboff — COMMENTED** [review]
@KerrMadeleine - Nice job, just a few comments to address!

**Comment by KerrMadeleine** [comment]
Here are the figures (Temperature, velocity and temperature across the mid mantle) that the script produces.
T_dev_00025
vphi_00025
vr_00025
mid_mantle_t_00025

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+    )
+    pl.add_mesh(midmantle_contour, color="red", line_width=2)
+    pl.view_xy()
+    pl.screenshot(dir + "plots/T_dev_" + num_str + ".png")
+    pl.show
+    pl.clear()
+    pl.close()
+
+
+#   A model to run that can utalize this scripts can be created by modifying
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:188]:
```suggestion
#   ASPECT model output that can utilize this script can be created
```

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 8]
```diff
@@ -0,0 +1,240 @@
...
+#   the onset-of-convection cookbook at:
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:189]:
```suggestion
#  by making the following changes to the onset-of-convection cookbook:
```

File: contrib/python/scripts/pyvista_2D_annulus_example.py [hunk 9]
```diff
@@ -0,0 +1,240 @@
...
+#   https://aspect-documentation.readthedocs.io/en/latest/user/cookbooks/cookbooks/onset_of_convection/doc/onset_of_convection.html
+#   and making the following changes to the parameter file: (< is original file, and > is modified version)
```
Comment (naliboff) [inline contrib/python/scripts/pyvista_2D_annulus_example.py:191]: remove this line

**Review by naliboff — CHANGES_REQUESTED** [review]
A final round of comments, but nearly there!

**Review by naliboff — APPROVED** [review]

**Comment by bangerth** [comment]
@KerrMadeleine This is quite nice, but it's going to be hard to find for people because it's not mentioned in any of the cookbooks. Since you already have the pictures, have you considered whether this could all be put into a short cookbook, right after the "onset of convection" cookbook, that shows the same thing in the spherical shell geometry you use here, and where both the modified input file could then be stored in the cookbook directory, and you could talk about the script you use here? Together with the pictures you already have, this would probably not be more than a page or two of text to write, but be quite useful to others!

**Comment by KerrMadeleine** [comment]
Yes, I could definitely make something like that. Thank you for suggesting.

**Comment by bangerth** [comment]
That would be fantastic! Let any of us know if you'd like help!

**Pull request merged**